src/database.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Provide the DatabaseObserver class to handle database relation and state."""
+
+import typing
+
+from charms.data_platform_libs.v0.data_interfaces import DatabaseRequires
+from ops.charm import CharmBase
+from ops.framework import Object
+
+DATABASE_NAME = "discourse"
+
+
+class DatabaseObserver(Object):
```

Comment:
Perhaps work with @gregory-schiano to clarify how to use observers? Or perhaps with @amandahla who has recently implemented it on Synapse: https://github.com/canonical/synapse-operator/blob/main/src/database_observer.py

Reply:
Answered with https://github.com/canonical/discourse-k8s-operator/pull/79#discussion_r1274056536

Reply:
It is a bit confusing to call this an observer if we don't intend to actually implement the pattern?

Reply:
Renamed to "handler"